### PR TITLE
Add Component Types INNER_SOURCE and SERVICE and rename existing types

### LINKED ISSUE #350: Component type `service` also for BOM needed.
Modern Apps tend to use services which are provided like openstreetmap, openweather.... the serivces, which are used by an App (i.e. project) shall be listed in sw360, because many of them do have terms of use and OSS involved. Perhaps it is ok to simply add a new `component type  to component which is `Service`.

### PR DESCRIPTION
**Pull request by alex-evo**
- add component types INNER_SOURCE and SERVICE 
- shorten display names of the existing component types

closes #350

### CODE REVIEW

**Comment by maximilianhuber** [comment]
The shorter display names might be harder to understand and use. Maybe add hovertexts or descriptions (later)?

File: libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/common/ThriftEnumUtils.java [hunk 1]
```diff
@@ -48,10 +48,11 @@ private ThriftEnumUtils() {
     );
     //! Enumeration "fancy name" mappings
     private static final ImmutableMap<ComponentType, String> MAP_COMPONENT_TYPE_STRING = ImmutableMap.of(
-            ComponentType.OSS, "Open Source Software" ,
-            ComponentType.COTS, "Commercial Off The Shelf" ,
-            ComponentType.INTERNAL, "Internal" ,
-            ComponentType.FREESOFTWARE, "Free Software");
```
Comment (mcjaeger) [inline libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/common/ThriftEnumUtils.java]: maybe still 'inner source'

Comment (alex-evo) [inline libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/common/ThriftEnumUtils.java]: I don't have a preference. I can see how Internal and Inner Source could be more familiar to Siemens employees.
@maximilianhuber , does Bosch have a name for this kind of components (Inner Source)?

**Comment by mcjaeger** [comment]
how about these types?

| before | after |
|---|---|
| OSS  | OSS  |
| COTS  | COTS  |
| Free Software  | Freeware  |
| Internal  | Internal |
| - | Inner Source |
| - | Service |

File: libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/common/ThriftEnumUtils.java (hunk 1 above)
Comment (maximilianhuber) [inline libraries/lib-datahandler/src/main/java/org/eclipse/sw360/datahandler/common/ThriftEnumUtils.java]: No, I do not think so

**Comment by alex-evo** [comment]
Updated the component types and texts

**Comment by mcjaeger** [comment]
actually, the tyoe does not seem to be displayed properly in the UI:
1. creating a servcie entry
screen shot 2017-04-28 at 16 18 38
2. displays empty in the listing:
screen shot 2017-04-28 at 16 18 51

**Comment by alex-evo** [comment]
That's a different can of worms. You assigned a type to a component, but you're looking at a release. I know, we talked about this already, but this feature is not yet implemented. That is, the component type is not displayed in the linked releases view. I can't seem find the issue, where this is requested.

**Comment by mcjaeger** [comment]
OK, then let's have this done when solving the issue #287

**Pull request merged**